### PR TITLE
Filename matcher error handling

### PR DESCRIPTION
**Pull request by angiebecerra020**
This PR improves the error handling mechanism for filename matching and enhances the overall stability of the tool by preventing unhandled crashes.

Changes Introduced:
✅ Added a custom exception FilenameMatcherError to handle errors when calling filename_matcher.
✅ Wrapped key functions (do, run, test) in try-except blocks to catch and log errors gracefully.
✅ Improved logging for better debugging and visibility of issues.
✅ Introduced a new test_filename_matcher command to verify filename matching logic.
✅ Updated test command to properly handle errors and exit with an appropriate failure status.

Why This Change?
Previously, unhandled exceptions in filename_matcher could cause crashes, making debugging difficult. This update improves robustness by ensuring errors are logged and handled properly.

How to Test?
Run python main.py test some_codemod.py and verify that filename matching errors are properly caught and displayed.
Try using invalid inputs for filename matching to see how errors are handled.
Run python main.py test_filename_matcher to validate the new testing command.
Impact & Risk Assessment:
🚀 Improves Stability: Prevents silent crashes and enhances debugging experience.
📢 Better Logging: Users get clearer feedback on errors.
✅ Backwards Compatible: No breaking changes introduced.

Would love feedback on any additional edge cases to test! 🙌

### CODE REVIEW

File: bowler/tool.py [hunk 1]
```diff
@@ -200,10 +200,21 @@ def refactor_dir(self, dir_name: str, *a, **k) -> None:
...
-                if not name.startswith(".") and self.filename_matcher(
-                    Filename(fullname)
-                ):
-                    self.queue_work(Filename(fullname))
+    # Skip hidden files (those starting with a dot)
+            
+max_retries = 3
+retries = 0
+
+while retries < max_retries:
```
Comment (thatch) [inline bowler/tool.py:208]: Something seems wrong with the indentation on this block, it displays at the left margin for me.

File: bowler/query.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import logging
 import pathlib
 import re
+import time # added by Angie Becerra
```
Comment (thatch) [inline bowler/query.py:12]: I don't think you need this line at all?
